### PR TITLE
Theming initialization should not rely on side effects to define load order

### LINKED ISSUE #8242: Remove "sideeffects: true" from styling
### Environment Information 
 
- __Package version(s)__: 6.152.1 
- __Browser and OS versions__: (Chrome 72, Windows 10)
`import * as React from "react";`
`import { Component } from "react";`
`import { render } from "react-dom";`
`import { Fabric } from 'office-ui-fabric-react/lib/Fabric';`
`export default class App`
`  	extends Component`
`{`
` 	render()`
`  	{`
`  		return <Fabric>WOHOO!!!</Fabric>;`
`  	}`
`}`
`const root = document.createElement("div");`
`root.id = "root";`
`document.body.appendChild(root);`
`render(<App />, root);`

 
#### Actual behavior:
Builds without error, doesn't render, Browser console shows following error:

TypeError: Cannot read property 'disableGlobalClassNames' of undefined
    at getGlobalClassNames.js:29
    at we (Fabric.styles.js:8)
    at ke (classNamesFunction.js:17)
    at t.render (Fabric.base.js:23)
    at t.e.(:1337/anonymous function) [as render] (http://localhost:1337/dist/main.js:46:24767)
    at Oo (react-dom.production.min.js:173)
    at No (react-dom.production.min.js:172)
    at Mo (react-dom.production.min.js:180)
    at Qa (react-dom.production.min.js:232)
    at Ka (react-dom.production.min.js:233)

#### Expected behavior:
 
 Builds without errors, displays WOHOO!!! in browser on render.
 
#### Priorities and help requested:
 
Are you willing to submit a PR to fix? Yes
 
Requested priority: High

### PR DESCRIPTION
**Pull request by dzearing**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #8242 
- [X] Include a change request file using `$ npm run change`

#### Description of changes

While we did fix sideEffects to include the theme.js page in a previous update to the `@uifabric/styling` package, I want to ensure that we have an explicit dependency on initialization. If you references the `{ DefaultButton } from 'oufr/lib/Button'`, that graph should return a function which explicitly imports and executes `{ initializeDefaultTheme } from '@uifabric/styling'`.

#### Focus areas to test

Need to bundle minified website to make sure minifying still works great.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8271)

### CODE REVIEW

**Comment by dzearing** [comment]
Investigating the bundle increases. If I bundle ProgressIndicator in isolation, the bundle size is smaller than if I build it along with all the other entries. So this seems like a webpack bug.

@TheLarkInn fyi. I attempted to remove the dependency on a side effect, and it ended up making things much worse

**Comment by aftab-hassan** [comment]
@phkuo, to add to @dzearing's comment above about webpack sizes, I have verified the size auditor logs. The bundle sizes seem to be computed correctly and the numbers reported are from the webpack run. 

Could you please take a look at this `PR`  ?
sizecomparison-theminginit

File: packages/styling/src/styles/theme.ts [hunk 1]
```diff
@@ -1,33 +1,48 @@
...
+
+    if (!_theme) {
+      const win = getWindow();
+
+      // tslint:disable:no-string-literal no-any
+      if (win && (win as any)['FabricConfig'] && (win as any)['FabricConfig'].theme) {
+        _theme = createTheme((win as any)['FabricConfig'].theme);
+      } else {
+        _theme = createTheme({
+          palette: DefaultPalette,
```
Comment (phkuo) [inline packages/styling/src/styles/theme.ts]: why is this necessary? doesn't createTheme() already populate all the defaults for you?

Comment (dzearing) [reply packages/styling/src/styles/theme.ts]: This takes the initialization into something that can be explicitly imported.

Comment (aftab-hassan) [reply packages/styling/src/styles/theme.ts]: @phkuo, @dzearing had a response to your question. Could you please give this another look and review?

**Comment by aftab-hassan** [comment]
@dzearing, while @phkuo is taking a look, looks like we have 1 merge conflict.

**Comment by phkuo** [comment]
Please have @dzearing review, he can bypass the code owner review restriction on me to check in when complete.

**Comment by dzearing** [comment]
Closing for now.
